packages/@ember/engine/index.ts

```diff
@@ -472,6 +472,9 @@ export function buildInitializerMethod<
   B extends 'initializers' | 'instanceInitializers',
   T extends B extends 'initializers' ? Engine : EngineInstance,
 >(bucketName: B, humanName: string) {
+  if (bucketName === '__proto__' || bucketName === 'constructor' || bucketName === 'prototype') {
```

Comment:
per the types (and usage in the codebase), `bucketName` is supposed to only be `initializers` or `instanceIntitializers`

```suggestion
  if (bucketName !== 'initializers' && bucketName !== 'instanceInitializers') {
```